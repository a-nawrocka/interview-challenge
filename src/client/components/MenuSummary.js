import React from "react";

class MenuSummary extends React.Component {
 
	render() {   
    let dietaries = {};

    this.props.selectedDishes.forEach(dish => {
      dish.dietaries.forEach(dietary => {
        dietaries[dietary] = (dietaries[dietary] ?? 0) + 1;
      });
    });

    let dietariesLabels = [];

    Object.entries(dietaries).map((entry,index) => {
      dietariesLabels.push(`${entry[1]}x `)
      dietariesLabels.push(<span key={index} className="dietary">{entry[0]}</span>)
    });

    return  (
      <div className="menu-summary">
          <div className="container">
            <div className="row">
              <div className="col-6 menu-summary-left">
                <span>{this.props.selectedDishes.length} {this.props.selectedDishes.length === 1 ? 'item' : 'items'}</span>
              </div>
              <div className="col-6 menu-summary-right">
                {dietariesLabels}
              </div>
            </div>
          </div>
      </div>
	  );
  };
};

export default MenuSummary;