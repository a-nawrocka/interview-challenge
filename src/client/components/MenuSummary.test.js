import React from "react";
import renderer from "react-test-renderer";
import MenuSummary from "./MenuSummary";

describe("Given valid array of selected dish objects", () => {
	it("displays summary of selected dishes", () => {
		const expectedDishes = [
			{
				id: 1,
				name: "dish one",
				dietaries: ["v", "ve", "df"],
			},
			{
				id: 2,
				name: "dish two",
				dietaries: ["rsf", "n!"],
			},
		];

		const component = renderer.create(<MenuSummary selectedDishes={expectedDishes} />);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
