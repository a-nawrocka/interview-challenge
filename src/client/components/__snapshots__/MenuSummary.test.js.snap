// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given valid array of selected dish objects displays summary of selected dishes 1`] = `
<div
  className="menu-summary"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-6 menu-summary-left"
      >
        <span>
          2
           
          items
        </span>
      </div>
      <div
        className="col-6 menu-summary-right"
      >
        1x 
        <span
          className="dietary"
        >
          v
        </span>
        1x 
        <span
          className="dietary"
        >
          ve
        </span>
        1x 
        <span
          className="dietary"
        >
          df
        </span>
        1x 
        <span
          className="dietary"
        >
          rsf
        </span>
        1x 
        <span
          className="dietary"
        >
          n!
        </span>
      </div>
    </div>
  </div>
</div>
`;
